staging/kos/pkg/controllers/connectionagent/connectionagent.go

```diff
@@ -0,0 +1,1059 @@
+type ConnectionAgent struct {
+	localNodeName string
+	hostIP        gonet.IP
+	kcs           *kosclientset.Clientset
+	netv1a1Ifc    netvifc1a1.NetworkV1alpha1Interface
+	queue         k8sworkqueue.RateLimitingInterface
+	workers       int
+	netFabric     netfabric.Interface
+	stopCh        <-chan struct{}
+
+	localAttsInformer k8scache.SharedIndexInformer
+	localAttsLister   koslisterv1a1.NetworkAttachmentLister
+
+	vniToVnStateMutex sync.RWMutex
+	vniToVnState      map[uint32]*vnState
+
+	localAttIfcsMutex sync.RWMutex
+	localAttIfcs      map[k8stypes.NamespacedName]netfabric.NetworkInterface
+
+	remoteAttIfcsMutex sync.RWMutex
+	remoteAttIfcs      map[k8stypes.NamespacedName]netfabric.NetworkInterface
+}
+
+func NewConnectionAgent(localNodeName string,
+	hostIP gonet.IP,
+	kcs *kosclientset.Clientset,
+	queue k8sworkqueue.RateLimitingInterface,
+	workers int,
+	netFabric netfabric.Interface) *ConnectionAgent {
+
+	return &ConnectionAgent{
+		localNodeName: localNodeName,
+		hostIP:        hostIP,
+		kcs:           kcs,
+		netv1a1Ifc:    kcs.NetworkV1alpha1(),
+		queue:         queue,
+		workers:       workers,
+		netFabric:     netFabric,
+		vniToVnState:  make(map[uint32]*vnState),
+		localAttIfcs:  make(map[k8stypes.NamespacedName]netfabric.NetworkInterface),
+		remoteAttIfcs: make(map[k8stypes.NamespacedName]netfabric.NetworkInterface),
+	}
+}
+
+func (ca *ConnectionAgent) Run(stopCh <-chan struct{}) error {
+	defer k8sutilruntime.HandleCrash()
+	defer ca.queue.ShutDown()
+
+	ca.stopCh = stopCh
+	ca.initLocalAttsInformerAndLister()
+	go ca.localAttsInformer.Run(stopCh)
+	glog.V(2).Infoln("Local NetworkAttachments Informer started")
+
+	err := ca.waitForLocalAttsCacheSync(stopCh)
+	if err != nil {
+		return err
+	}
+	glog.V(2).Infoln("Local NetworkAttachments cache synced")
+
+	err = ca.handlePreExistingLocalIfcs()
+	if err != nil {
+		// ? An error is returned if deleting a single interface fails. This might be a bad idea: we might just want to keep running
+		// (after logging the failure) if out of 1k interfaces one could not be deleted. In fact, if the network fabric (OvS) deletion
+		// can occasionally experience transient failures and we have a lot of interfaces on the node, the fact that we are just returning
+		// an error might make it impossible to make the connection agent successfully start up. If that's the case (OvS can experience
+		// flaky failures), we should retry to delete the interface after some time, not just simply log the failure, because that would
+		// mean leaving on the node an interface the connection agent is not aware of, and this could cause troubles if later an attachment
+		// which needs an interface with the same MAC as the one which could not be deleted is created. If insterad OvS is reliable and fails
+		// only when there's something wrong with the system state, then the current solution is fine. Choose what to do after the OvS network
+		// fabric is ready.
```

Comment:
I think an even simpler solution would be practical.  If a fabric call fails due to a communication failure then wait a little while and retry.  That is, do not enter normal operation until OVS has recovered from whatever comm problem it had.

Reply:
Suggestion implemented in 
dfcd8d4.